Python/pywarpx/_libwarpx.py

```diff
@@ -561,6 +563,58 @@ def get_particle_arrays(species_name, comp_name, level):
     _libc.free(data)
     return particle_data
 
+def get_particle_boundary_buffer(species_name, boundary, comp_name, level):
+    '''
+
+    This returns a list of numpy arrays containing the particle array data
+    for a species that has been scraped by a specific simulation boundary.
+
+    The data for the numpy arrays are not copied, but share the underlying
+    memory buffer with WarpX. The numpy arrays are fully writeable.
+
+    Parameters
+    ----------
+
+        species_name   : the species name that the data will be returned for.
+        boundary       : the boundary from which to get the scraped particle data.
+                         In the form x/y/z_hi/lo
+        comp_name      : the component of the array data that will be returned.
+        level          : Which AMR level to retrieve scraped particle data from.
+    Returns
+    -------
+
+        A List of numpy arrays.
+
+    '''
+    dimensions = {'x' : 1, 'y' : 2, 'z' : 3}
```

Comment:
I think this should be something like:
```suggestion
    if self.geometry_dim == '3d':
        dimensions = {'x' : 1, 'y' : 2, 'z' : 3}
    elif self.geometry_dim == '2d':
        dimensions = {'x' : 1, 'z' : 2}
```

Reply:
ah true, forgot about that difference